lib/rubysl/rubysl-socket/lib/rubysl/socket/foreign.rb

```diff
@@ -222,6 +222,17 @@ def self.getsockname(descriptor)
 
       def self.pack_sockaddr_in(host, port, family = ::Socket::AF_UNSPEC,
                                 type = 0, flags = 0)
+        if type == 0
+          begin
+            # Check if the port is a numeric value we must set the socket type in order for the getaddrinfo call
+            # to succeed on some platforms.
```

Comment:
It was a bug in that it didn't match MRI behavior. I provided the context for it in a fairly thorough commit message.